os/arch/arm/src/armv8-m/up_assert.c

```diff
@@ -457,6 +457,10 @@ void up_assert(const uint8_t *filename, int lineno)
 {
 	board_led_on(LED_ASSERTION);
 
+#ifdef CONFIG_SYSTEM_REBOOT_REASON
+	reboot_reason_trywrite();
```

Comment:
Let's split commits for this change
1. add new api support on reboot reason
2. apply it on each chip